Propose Solution for connectorSizing (#3774)
modelica/ModelicaSpecification: chapters/annotations.tex

In that case the value modifier should not be edited directly by the user, and a tool may choose to not display that variable in the dialog or display it with disabled input field.
There shall be exactly one vector of connectors using the dimension size \lstinline!n! for each variable \lstinline!n! with \lstinline!connectorSizing = true!.
Other vectors of connectors that need to have the same size may use a different expression with the same value, such as \lstinline!n + 0!, to avoid that multiple vectors become associated with the \lstinline!connectorSizing! automatic updating of \lstinline!n!.
(Note that non-connector vectors may have size \lstinline!n! without becoming associated.)

\begin{nonnormative}
The \lstinline!connectorSizing! annotation is used in cases where connections to a vector of connectors shall be made and a new connection requires to resize the vector and to connect to the new index (unary connections).
The annotation allows a tool to perform these two actions in many cases automatically.
This is, e.g., very useful for state machines and for certain components of fluid libraries.

> If a variable \lstinline!n! with \lstinline!connectorSizing = true! has multiple associated vectors of connectors, some of the graphical operations described below will not work reliably.

Reviewer: Even better with _would_:

```suggestion
If a variable \lstinline!n! with \lstinline!connectorSizing = true! would have multiple associated vectors of connectors, some of the graphical operations described below would not work reliably.
```

There shall be exactly one vector of connectors using the dimension size \lstinline!n! for each variable \lstinline!n! with \lstinline!connectorSizing = true!.
Other vectors of connectors that need to have the same size may use a different expression with the same value, such as \lstinline!n + 0!, to avoid that multiple vectors become associated with the \lstinline!connectorSizing! automatic updating of \lstinline!n!.
(Note that non-connector vectors may have size \lstinline!n! without becoming associated.)

\begin{nonnormative}
The \lstinline!connectorSizing! annotation is used in cases where connections to a vector of connectors shall be made and a new connection requires to resize the vector and to connect to the new index (unary connections).
The annotation allows a tool to perform these two actions in many cases automatically.
This is, e.g., very useful for state machines and for certain components of fluid libraries.

If there is not exactly one vector of connectors for a each variable \lstinline!n! with \lstinline!connectorSizing = true! diagnostics are needed, but the model may otherwise work.

> If there are no vector of connectors for such a variable it should be removed.

Reviewer: Clarify what to remove:
```suggestion
The recommended action if there is no associated vector of connectors is to remove the \lstinline!connectorSizing! annotation.
```

Author: Oh, my intent was to remove the variable - as you have no way of modifying it.
```suggestion
The recommended action if there is no associated vector of connectors is to remove the \lstinline!connectorSizing! annotation or the variable declaration.
```

Reviewer: But without `connectorSizing` it would just be a normal parameter, and those are typically very easy to modify.  Maybe the parameter is also used for lots of non-connector vector sizes.

Author: That it was used a lot seems unlikely when users couldn't modify it through the GUI, so it was effectively stuck at 0. But I could change the text to "or possibly the variable declaration".

If \lstinline!connectorSizing = true!, a tool may set the variable value in a modifier automatically, if used as dimension size of a vector of connectors.
In that case the value modifier should not be edited directly by the user, and a tool may choose to not display that variable in the dialog or display it with disabled input field.

> There should be one vector of connectors using the dimension size \lstinline!n! for each variable \lstinline!n! with \lstinline!connectorSizing = true!.

Author: There wasn't one, so I can reformulate.

> There should be one vector of connectors using the dimension size \lstinline!n! for each variable \lstinline!n! with \lstinline!connectorSizing = true!.

Reviewer: OK.  Resolving.

> If there are other vector of connectors that should have the same size, but not cause an automatic update they can use \lstinline!n+0! instead.

Reviewer: Avoid use of _should_ here:

```suggestion
Other vectors of connectors that need to have the same size may use a different expression with the same value, such as \lstinline!n + 0!, to avoid that multiple vectors become associated with the \lstinline!connectorSizing! automatic updating of \lstinline!n!.
```
